train_segmentation.py

```diff
@@ -150,38 +173,66 @@ def calc_eval_batchsize(gpu_devices_dict: dict, batch_size: int, sample_size: in
-                raise logging.critical(
-                    IndexError(f"\nThe number of training samples in the configuration file ({num_samples[i]}) "
-                               f"exceeds the number of samples in the hdf5 training dataset ({file_num_samples}).")
-                )
+    for dataset in ['trn', 'val', 'tst']:
+        dataset_file, _ = Tiler.make_dataset_file_name(experiment_name, min_annot_perc, dataset, attr_vals)
+        dataset_filepath = samples_path / dataset_file
+        if not dataset_filepath.is_file() and dataset == 'tst':
+            num_samples[dataset] = 0
+            logging.warning(f"No test set. File not found: {dataset_filepath}")
+            continue
+
+        if get_key_def(f"num_{dataset}_samples", params['training'], None) is not None:
+            num_samples[dataset] = params['training'][f"num_{dataset}_samples"]
+
+            with open(dataset_filepath, 'r') as datafile:
+                file_num_samples = len(datafile.readlines())
+            if num_samples[dataset] > file_num_samples:
+                raise IndexError(f"The number of training samples in the configuration file ({num_samples[dataset]}) "
+                                 f"exceeds the number of samples in the hdf5 training dataset ({file_num_samples}).")
         else:
-            with h5py.File(samples_path.joinpath(f"{i}_samples.hdf5"), "r") as hdf5_file:
-                num_samples[i] = len(hdf5_file['map_img'])
-
-        with h5py.File(samples_path.joinpath(f"{i}_samples.hdf5"), "r") as hdf5_file:
-            if i == 'trn':
-                for x in range(num_samples[i]):
-                    label = hdf5_file['map_img'][x]
-                    unique_labels = np.unique(label)
-                    weights.append(''.join([str(int(i)) for i in unique_labels]))
-                    samples_weight = compute_sample_weight('balanced', weights)
+            with open(dataset_filepath, 'r') as datafile:
+                num_samples[dataset] = len(datafile.readlines())
+
+        with open(dataset_filepath, 'r') as datafile:
+            datalist = datafile.readlines()
+            if dataset == 'trn':
+                if not compute_sampler_weights:
+                    samples_weight = np.ones(num_samples[dataset])
+                else:
+                    dontcare = get_key_def("ignore_index", params['dataset'], default=-1)
+                    for x in tqdm(range(num_samples[dataset]), desc="Computing sample weights"):
+                        label_file = datalist[x].split(';')[1]
+                        with rasterio.open(label_file, 'r') as label_handle:
+                            label = label_handle.read()
+                        label = np.where(label == dontcare, 0, label)
+                        unique_labels = np.unique(label)
+                        weights.append(''.join([str(int(i)) for i in unique_labels]))
+                        samples_weight = compute_sample_weight('balanced', weights)
```

Comment:
@valhassan are you ok if we softcode this? It creates lots of overhead at train time and for a binary task with patches already filtered to discard "background only" patches, I think we can skip the weighting component during training. It is set to False by default, but can be used by set training.compute_sampler_weights=True in config/training.

I quickly tested the weighting from .tif patches and it seems to be working fine.

Reply:
patched it in a separate branch similar to what you have. Works!
code snippet attached for reference
![image](https://user-images.githubusercontent.com/24499815/199634127-1cdfc3a8-ce49-4252-a660-835349447391.png)